pkgs/development/tools/click/default.nix

```
@@ -0,0 +1,92 @@
+{ lib
+, fetchFromGitLab
+, buildPythonApplication
+, autoreconfHook
+, debian
+, automake
+, perl
+, libtool
+, vala
+, pkg-config
+, libgee
+, json-glib
+, properties-cpp
+, gobject-introspection
+, getopt
+, setuptools
+, pygobject3
+, wrapGAppsHook
+}:
+
+buildPythonApplication {
+  pname = "click";
+  version = "22-02-2023";
+
+  src = fetchFromGitLab {
+    owner = "ubports";
+    repo = "development/core/click";
+    rev = "aaf2735e8e6cbeaf2e429c70136733513a81718a";
+    sha256 = "sha256-pNu995/w3tbz15QQVdVYBnWnAoZmqWj1DN/5PZZ0iZw=";
+  };
+
+  configureFlags = [
+    "--with-systemdsystemunitdir=${placeholder "out"}/lib/systemd/system"
+    "--with-systemduserunitdir=${placeholder "out"}/usr/lib/systemd/user"
+  ];
+
+  preConfigure = ''
+    export click_cv_perl_vendorlib=$out/${perl.libPrefix}
+    export PYTHON_INSTALL_FLAGS="--prefix=$out"
+  '';
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r {bin,click_package,build,lib} $out
+    # cp -r . $out
+  '';
+
+  preFixup = ''
+    makeWrapperArgs+=(--prefix GI_TYPELIB_PATH : "${placeholder "out"}/lib/click" --prefix LD_LIBRARY_PATH : "${placeholder "out"}/lib")
```

1. `GI_TYPELIB_PATH` value is wrong, `wrapGAppsHook` sets this to `gobject-introspection>/lib/girepository-1.0:<libgee>/lib/girepository-1.0:<json-glib>/lib/girepository-1.0:<click>/lib/girepository-1.0`
2. Redundant since you added `wrapGAppsHook`, which will correctly apply this wrapping on its own. Only `LD_LIBRARY_PATH` is necessary here.

```suggestion
    makeWrapperArgs+=(--prefix LD_LIBRARY_PATH : "${placeholder "out"}/lib")
```